src/Parallel/Tags/ResourceInfo.hpp

```diff
@@ -63,5 +67,36 @@ struct SingletonInfo : db::SimpleTag {
     return resource_info.template get_singleton_info<ParallelComponent>();
   }
 };
+
```

Comment:
why not just add a single tag for a `ResourceInfo` and store that in the global cache?